### PR TITLE
Add additional data field to user entity

### PR DESCRIPTION
**Pull request by himanshu-dixit**

### CODE REVIEW

File: src/SocialAuthUserManager.php [hunk 1]
```diff
@@ -162,7 +162,7 @@ public function setSessionKeysToNullify(array $session_keys) {
    * @return \Symfony\Component\HttpFoundation\RedirectResponse
    *   A redirect response.
    */
-  public function authenticateUser($name, $email, $pluginId, $provider_user_id, $picture_url = FALSE) {
+  public function authenticateUser($name, $email, $pluginId, $provider_user_id, $picture_url = FALSE, $data) {
```
Comment (gvso) [inline src/SocialAuthUserManager.php]: Argument with default value must be at the end.

File: src/SocialAuthUserManager.php [hunk 2]
```diff
@@ -189,7 +189,7 @@ public function authenticateUser($name, $email, $pluginId, $provider_user_id, $p
       // Check if User with same email account exists.
       if ($drupal_user) {
         // Add record for the same user.
-        $this->addUserRecord($drupal_user->id(), $pluginId, $provider_user_id);
+        $this->addUserRecord($drupal_user->id(), $pluginId, $provider_user_id, $data);
```
Comment (gvso) [inline src/SocialAuthUserManager.php:205]: You're not passing $data in line 172. It should be something like this.

File: src/SocialAuthUserManager.php [hunk 3]
```diff
@@ -333,14 +333,10 @@ public function checkIfUserExists($pluginId, $provider_user_id) {
...
       ->execute();
-
-    $user_data = $storage->load(reset($social_auth_user));
-
-    if ($social_auth_user) {
-      // Return User ID.
-      return $user_data->get('user_id')->getValue()[0]['value'];
-    }
-    return FALSE;
+    if (!$social_auth_user)
```
Comment (gvso) [inline src/SocialAuthUserManager.php]: You must enclose the block in brackets even if the block contains only one line of code.

File: src/SocialAuthUserManager.php [hunk 4]
```diff
@@ -353,7 +349,7 @@ public function checkIfUserExists($pluginId, $provider_user_id) {
    * @param string $provider_user_id
    *   Unique Social ID returned by social network.
    */
-  public function addUserRecord($user_id, $pluginId, $provider_user_id) {
+  public function addUserRecord($user_id, $pluginId, $provider_user_id, $user_data) {
```
Comment (gvso) [inline src/SocialAuthUserManager.php]: You must add return tag in method documentation.

File: src/SocialAuthUserManager.php [hunk 5]
```diff
@@ -363,7 +359,7 @@ public function addUserRecord($user_id, $pluginId, $provider_user_id) {
           array(
```
Comment (gvso) [inline src/SocialAuthUserManager.php]: array() syntax must be replaced by [] due to a new coding standard. Please change all the matches in this file.

[Leave the code cleaner than you found it](http://pragmaticcraftsman.com/2011/03/the-boy-scout-rule/)

File: src/Entity/SocialAuth.php [hunk 6]
```diff
@@ -20,7 +20,8 @@
  *     "id" = "id",
```
Comment (gvso) [inline src/Entity/SocialAuth.php:20]: Please run PHP CodeSniffer before any commits.

**Review by gvso — CHANGES_REQUESTED** [review]

File: src/Entity/SocialAuth.php (hunk 6 above)
Comment (himanshu-dixit) [reply src/Entity/SocialAuth.php:20]: Is there a problem with this file? PHPCS is not showing any violation for this particular file. Or were you making a general comment.

File: src/SocialAuthUserManager.php (hunk 2 above)
Comment (himanshu-dixit) [reply src/SocialAuthUserManager.php:205]: Done.

File: src/SocialAuthUserManager.php [hunk 7]
```diff
@@ -490,7 +507,7 @@ public function loginUser(User $drupal_user) {
   protected function nullifySessionKeys() {
     if (!empty($this->sessionKeys)) {
       array_walk($this->sessionKeys, function ($session_key) {
-        $this->session->set($session_key, NULL);
+        $this->session->set($this->dataHandler->getSessionPrefix() . $session_key, NULL);
```
Comment (himanshu-dixit) [inline src/SocialAuthUserManager.php:510]: I'm still confused with this. I still think I might be making a mistake. Please, can you check if this part is correct?

**Review by himanshu-dixit — COMMENTED** [review]
@GVSO I have addressed your review comments, Can you check it once again?

File: src/Entity/SocialAuth.php (hunk 6 above)
Comment (gvso) [reply src/Entity/SocialAuth.php:20]: It was a general comment

File: src/SocialAuthUserManager.php [hunk 8]
```diff
@@ -156,20 +167,22 @@ public function setSessionKeysToNullify(array $session_keys) {
    *   The social implementer identifier.
    * @param string $provider_user_id
    *   The unique id returned by the user.
+   * @param JSON $data
```
Comment (gvso) [inline src/SocialAuthUserManager.php:170]: JSON is not a PHP data type. 
$data is probably either a (serialized) string or an array. Don't exactly know how this work yet

File: src/SocialAuthUserManager.php [hunk 9]
```diff
@@ -156,20 +167,22 @@ public function setSessionKeysToNullify(array $session_keys) {
...
+   *   The additional user_data to be stored in database.
    * @param string|bool $picture_url
    *   The user's picture.
    *
    * @return \Symfony\Component\HttpFoundation\RedirectResponse
    *   A redirect response.
    */
-  public function authenticateUser($name, $email, $pluginId, $provider_user_id, $picture_url = FALSE) {
+  public function authenticateUser($name, $email, $pluginId, $provider_user_id, $data, $picture_url = FALSE) {
```
Comment (gvso) [inline src/SocialAuthUserManager.php:178]: To make it backward compatible, I would put $data with a default value at the end

File: src/SocialAuthUserManager.php [hunk 10]
```diff
@@ -352,19 +362,25 @@ public function checkIfUserExists($pluginId, $provider_user_id) {
    *   Type of social network.
    * @param string $provider_user_id
    *   Unique Social ID returned by social network.
+   * @param JSON $user_data
```
Comment (gvso) [inline src/SocialAuthUserManager.php:365]: JSON data type doesn't exist

File: src/SocialAuthUserManager.php (hunk 7 above)
Comment (gvso) [reply src/SocialAuthUserManager.php:510]: You have a data handler object, you don't need a Symfony\Component\HttpFoundation\Session\Session object anymore. Instead, use the data handler to set, get, whatever data in session.

That being said, remove the $session service from this class

**Review by gvso — CHANGES_REQUESTED** [review]

**Comment by himanshu-dixit** [comment]
Closed, please see #5

**Pull request closed without merging**